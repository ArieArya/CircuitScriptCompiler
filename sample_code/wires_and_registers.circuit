reg r1 = 1;
reg r2 = 0;

wire w1 = or(r1, or(r1, r2));
wire w2 = and(r1, r2);

print(w1);
print(w2);

wire w3 = nand(w1, w2);
print(w3);