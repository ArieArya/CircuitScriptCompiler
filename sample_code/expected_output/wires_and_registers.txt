Lexing file: wires_and_registers.circuit
Tokens:
Token.KEYWORD (Lexeme = 'wire')
Token.IDENTIFIER (Lexeme = 'w1')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'wire')
Token.IDENTIFIER (Lexeme = 'w2')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'reg')
Token.IDENTIFIER (Lexeme = 'r1')
Token.OPERATOR (Lexeme = '=')
Token.DIGIT (Lexeme = '1')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'reg')
Token.IDENTIFIER (Lexeme = 'r2')
Token.OPERATOR (Lexeme = '=')
Token.DIGIT (Lexeme = '0')
Token.SEMICOLON (Lexeme = ';')
Token.IDENTIFIER (Lexeme = 'w1')
Token.OPERATOR (Lexeme = '=')
Token.KEYWORD (Lexeme = 'or')
Token.LPAREN (Lexeme = '(')
Token.IDENTIFIER (Lexeme = 'r1')
Token.COMMA (Lexeme = ',')
Token.IDENTIFIER (Lexeme = 'r2')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.IDENTIFIER (Lexeme = 'w2')
Token.OPERATOR (Lexeme = '=')
Token.KEYWORD (Lexeme = 'and')
Token.LPAREN (Lexeme = '(')
Token.IDENTIFIER (Lexeme = 'r1')
Token.COMMA (Lexeme = ',')
Token.IDENTIFIER (Lexeme = 'r2')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.IDENTIFIER (Lexeme = 'w1')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.IDENTIFIER (Lexeme = 'w2')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
